### ISSUE TITLE
CLion 2023.1.3, 2022.3.3 and IntelliJ idea community 2023.1.2 false positive "cannot move"

### ISSUE THREAD

**Issue by Qwinci** [intellij-rust/intellij-rust#10495]
## Environment

* **IntelliJ Rust plugin version:** 0.4.194.5382-231
* **Rust toolchain version:** 1.71.0-nightly (9d871b061 2023-05-21) x86_64-unknown-linux-gnu
* **IDE name and version:** CLion 2023.1.3 (CL-231.9011.31)
* **Operating system:** Linux 6.3.3-arch1-1
* **Macro expansion:** enabled

## Problem description
CLion incorrectly reports a Copy field as "cannot move". I tried it in both CLion 2023.1.3, 2022.3.3 and a fresh install of IntelliJ Idea Community Edition and all of them have the same issue.
![image](https://github.com/intellij-rust/intellij-rust/assets/32550582/bc3d35ec-1bc2-4c8c-b716-b2e4f823faff)

## Steps to reproduce
Create a new Rust project in CLion and add a struct with a Copy type field and access that in a member function.

**Comment by Belgabor** [comment]
Same issue here, cannot exactly identify when it happened since I haven't worked on my rust code in a few days, but I noticed it today after upgrading IDEA community edition from 2023.1.1 to 2023.1.2.

**Comment by tisonkun** [comment]
The same here. `u64` that impls `Copy` trait should not trigger this warning. Also, `cargo clippy` will correct if I try to suppress IDE complains with `.clone()`.

**Comment by ChristianIvicevic** [comment]
I can confirm this with the following reproduction which only happens with the nightly compiler. With the stable compiler there is no problems.
```rust
pub struct Point {
    pub x: i32,
    pub y: i32,
}

pub struct RectangleArea {
    pub left: i32,
    pub top: i32,
    pub right: i32,
    pub bottom: i32,
}

impl RectangleArea {
    #[must_use]
    pub fn inflate(&self, dx: i32, dy: i32) -> Self {
        Self {
            left: self.left - dx,
               // ^ cannot move self
            top: self.top - dy,
              // ^ cannot move self
            right: self.right + dx,
                // ^ cannot move self and dx
            bottom: self.bottom + dy,
                 // ^ cannot move self and dy
        }
    }
}
```

## Environment

* **IntelliJ Rust plugin version:** 0.4.196.5395-232-nightly
* **Rust toolchain version:** 1.71.0-nightly (8b4b20836 2023-05-22) aarch64-apple-darwin
* **IDE name and version:** CLion 2023.2 EAP (CL-232.6509)
* **Operating system:** macOS 13.2.1
* **Macro expansion:** enabled
* **Additional experimental features:** org.rust.macros.proc.attr, org.rust.cargo.emulate.terminal, org.rust.macros.proc

**Comment by itmuckel** [comment]
Only workaround so far probably is disabling `Borrow checker` inspection and rely on clippy for errors, but it feels wrong to do that.

**Comment by BeastLe9enD** [comment]
Same here, would be really neat to see this fixed :)

**Comment by mili-l** [comment]
Hi, thanks everyone, please see  https://github.com/intellij-rust/intellij-rust/issues/10427#issuecomment-1534673947

**Comment by mili-l** [comment]
Duplicated #10427
